lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1986,7 +1986,7 @@ sub investigate {
     return {error => 'No result directory available for current job'} unless $self->result_dir();
     my $ignore = OpenQA::App->singleton->config->{global}->{job_investigate_ignore};
     for my $prev (@previous) {
-        if ($prev->result eq 'failed') {
+        if ($prev->should_show_investigation) {
```

Comment:
this change causes tests to fail, don't know why. `$prev->result =~ /^failed$|incomplete/` would work but I would prefer to use the constants. But also I wonder if we should not change the investigate method to only look for a "first bad" which has the same result as the current job. Like, if we are looking for a previous job with the same problem, then it has to have the same result, otherwise it's a different problem, right?